sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import.service.ts

```diff
@@ -0,0 +1,127 @@
+import { Injectable } from "@nestjs/common";
+import {
+  ApplicationBulkWithdrawalHeader,
+  ApplicationWithdrawalTextValidationResult,
+} from "./application-bulk-withdrawal-import-text.models";
+import { DataSource, In, Repository } from "typeorm";
+import { Application } from "@sims/sims-db";
+import {
+  ApplicationBulkWithdrawalImportBusinessValidationModel,
+  ApplicationData,
+} from "./application-bulk-withdrawal-import-business-validation.models";
+import { InjectRepository } from "@nestjs/typeorm";
+
+type ApplicationDataMap = Record<string, ApplicationData>;
+
+/**
+ * Handles the application withdrawal bulk insert preparation.
+ */
+@Injectable()
+export class ApplicationBulkWithdrawalImportService {
+  constructor(
+    private readonly dataSource: DataSource,
+    @InjectRepository(Application)
+    private readonly applicationRepo: Repository<Application>,
+  ) {}
+
+  /**
+   * Generates the application bulk withdrawal business validation model.
+   * @param textValidations text validation models to be converted to the business validation models.
+   * @param textHeader text header model.
+   * @param institutionId institution id.
+   * @param allowedLocationIds set of allowed location ids for the given institution.
+   * @returns application bulk withdrawal business models to be validated and persisted.
+   */
+  async generateApplicationBulkWithdrawalValidationModels(
+    textValidations: ApplicationWithdrawalTextValidationResult[],
+    textHeader: ApplicationBulkWithdrawalHeader,
+    institutionId: number,
+    allowedLocationIds: number[],
+  ): Promise<ApplicationBulkWithdrawalImportBusinessValidationModel[]> {
+    const applicationDataMap: ApplicationDataMap = {};
+    const applicationNumbers = textValidations.map(
+      (textValidation) => textValidation.textModel.applicationNumber,
+    );
+    const applicationData =
+      await this.getApplicationBulkWithdrawalValidationDetails(
+        applicationNumbers,
+        institutionId,
+      );
+    applicationData.forEach((record) => {
+      applicationDataMap[record.applicationNumber] = {
+        applicationStatus: record.applicationStatus,
+        isArchived: record.isArchived,
+        sin: record.student.sinValidation.sin,
+        locationId: record.location.id,
+        locationCode: record.location.institutionCode,
+        hasPreviouslyBeenWithdrawn: !!record.studentScholasticStandings.length,
+      };
+    });
+    const businessValidationModels: ApplicationBulkWithdrawalImportBusinessValidationModel[] =
+      [];
+    textValidations.forEach((textValidation) => {
+      const businessValidationModel =
+        {} as ApplicationBulkWithdrawalImportBusinessValidationModel;
+      const applicationData =
```

Comment:
`applicationData` name is already used in line 45, pls use a different variable

Reply:
hmm... but the scope of this one in line 65 is limited to the callback function, so it shouldn't be a problem. However, to keep clarity, I am renaming the one in line 45 to `applicationValidationData`.